org.adempiere.base/src/org/adempiere/impexp/GridTabCSVImporter.java

```diff
@@ -684,7 +684,14 @@ private String processRecord(String importMode, GridTab gridTab, int indxDetail,
 						PO po = currentGridTab.getTableModel().getPO(currentGridTab.getCurrentRow());		
 						//Keep master record for details validation 
 						if(currentGridTab.equals(gridTab))
+						{
 							masterRecord = po;
+						}
+						else
+						{
+							if (currentGridTab.getParentTab() != null)
+								currentGridTab.getParentTab().dataRefresh(false);

```

Comment:
I think this will add significant overhead if there are many lines to import. Will be good if we can batch the refresh of the parent tab instead.

Reply:
hi @CarlosRuiz-globalqss , are you ok with the performance overhead added ? If you are fine with it, I guess we can proceed to merge this.

Reply:
@hengsin I retested with master branch and not able to reproduce issue. So I think it is fine if we do not apply this at time. If any one able to reproduce later, then we can use this PR.